samples/python/hello_classification/hello_classification.py

```diff
@@ -2,121 +2,111 @@
 
-    # Get a number of classes recognized by a model
-    num_of_classes = max(net.outputs[out_blob].shape)
+    _, h, w, _ = input_tensor.shape
 
-    # ---------------------------Step 4. Loading model to the device-------------------------------------------------------
-    log.info('Loading the model to the plugin')
-    exec_net = ie.load_network(network=net, device_name=args.device)
+    # 1) Set input tensor information:
+    # - input() provides information about a single model input
+    # - precision of tensor is supposed to be 'u8'
+    # - layout of data is 'NHWC'
+    # - set static spatial dimensions to input tensor to resize from
+    ppp.input().tensor() \
+        .set_element_type(Type.u8) \
+        .set_layout(Layout('NHWC')) \
+        .set_spatial_static_shape(h, w)  # noqa: ECE001, N400
 
-    # ---------------------------Step 5. Create infer request--------------------------------------------------------------
-    # load_network() method of the IECore class with a specified number of requests (default 1) returns an ExecutableNetwork
-    # instance which stores infer requests. So you already created Infer requests in the previous step.
+    # 2) Adding explicit preprocessing steps:
+    # - apply linear resize from tensor spatial dims to model spatial dims
+    ppp.input().preprocess().resize(ResizeAlgorithm.RESIZE_LINEAR)
 
-    # ---------------------------Step 6. Prepare input---------------------------------------------------------------------
-    original_image = cv2.imread(args.input)
-    image = original_image.copy()
-    _, _, h, w = net.input_info[input_blob].input_data.shape
+    # 3) Here we suppose model has 'NCHW' layout for input
+    ppp.input().model().set_layout(Layout('NCHW'))
 
-    if image.shape[:-1] != (h, w):
-        log.warning(f'Image {args.input} is resized from {image.shape[:-1]} to {(h, w)}')
-        image = cv2.resize(image, (w, h))
+    # 4) Set output tensor information:
+    # - precision of tensor is supposed to be 'f32'
+    ppp.output().tensor().set_element_type(Type.f32)
 
-    # Change data layout from HWC to CHW
-    image = image.transpose((2, 0, 1))
-    # Add N dimension to transform to NCHW
-    image = np.expand_dims(image, axis=0)
+    # 5) Apply preprocessing modifing the original 'model'
+    model = ppp.build()
 
-    # ---------------------------Step 7. Do inference----------------------------------------------------------------------
+# --------------------------- Step 5. Loading model to the device -----------------------------------------------------
+    log.info('Loading the model to the plugin')
+    compiled_model = core.compile_model(model, device_name)
+
+# --------------------------- Step 6. Create infer request and do inference synchronously -----------------------------
     log.info('Starting inference in synchronous mode')
-    res = exec_net.infer(inputs={input_blob: image})
+    output_tensor = compiled_model.infer_new_request({0: input_tensor})
```

Comment:
I think name `results` is better as later on you iter over it. If you were to iter over `output_tensor` you would access specific values of it.

As well you can use raw numpy array as input for this function. That will require some refactoring above.

Reply:
done